s4e-web/src/app/views/settings/manage-institutions/parent-institution-modal/parent-institution-modal.component.ts

```diff
@@ -0,0 +1,72 @@
+import { InstitutionService } from './../../state/institution.service';
+import { untilDestroyed } from 'ngx-take-until-destroy';
+import { InstitutionQuery } from './../../state/institution.query';
+import { ModalComponent } from './../../../../modal/utils/modal/modal.component';
+import { Component, Inject, OnDestroy, OnInit } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { ModalService } from 'src/app/modal/state/modal.service';
+import { MODAL_DEF } from 'src/app/modal/modal.providers';
+import { assertModalType } from 'src/app/modal/utils/modal/misc';
+import {
+  isParentInstitutionModal,
+  ParentInstitutionModal,
+  PARENT_INSTITUTION_MODAL_ID
+} from './parent-institution-modal.model';
+import { Institution } from '../../state/institution.model';
+import { FormControl } from '@ng-stack/forms';
+
+@Component({
+  selector: 's4e-parent-institution-modal',
+  templateUrl: './parent-institution-modal.component.html',
+  styleUrls: ['./parent-institution-modal.component.scss']
+})
+export class ParentInstitutionModalComponent extends ModalComponent<Institution> implements OnInit, OnDestroy {
+  public institutions: Institution[] = [];
+  public searchedInstitutions: Institution[] = [];
+  public selectedInstitution: Institution = null;
+  public institutionsSearch: FormControl<string> = new FormControl<string>('');
+
+  constructor(
+    private _http: HttpClient,
+    private _modalService: ModalService,
+    private _institutionQuery: InstitutionQuery,
+    private _institutionService: InstitutionService,
+    @Inject(MODAL_DEF) modal: ParentInstitutionModal
+  ) {
+    super(_modalService, PARENT_INSTITUTION_MODAL_ID);
+
+    assertModalType(isParentInstitutionModal, modal);
+  }
+
+  ngOnInit() {
+    this._institutionService.get();
+    this._institutionQuery.selectAll()
+      .pipe(untilDestroyed(this))
+      .subscribe(institutions => {
+        this.institutions = institutions;
+        this.searchedInstitutions = this.institutions;
+      });
+
+    this.institutionsSearch.valueChanges
+      .pipe(untilDestroyed(this))
+      .subscribe(() => {
+        this.searchedInstitutions = this.institutions
+          .filter(institution => this._isSearchedInstitution(institution));
+      });
```

Comment:
I would suggest connecting these two observables into one. The race condition can in theory occur here.

The workflow should be that you should `combineLatest` from search query and institutions.